YoCode/DuplicationCheck.cs

```diff
@@ -59,6 +70,57 @@ public DuplicationCheck(IPathManager dir, IDupFinder dupFinder)
             DuplicationEvidence.GiveEvidence(origEvidence, modEvidence);
         }
 
+        private void CheckForSpecialRepetition()
+        {
+            var csUris = dir.GetFilesInDirectory(dir.modifiedTestDirPath,FileTypes.cs);
+
+            var csUrisWithoutUnitTests = csUris.Where(a => !a.Contains("UnitConverterTests")).ToList();
+
+            var htmlUris = dir.GetFilesInDirectory(dir.modifiedTestDirPath, FileTypes.html).ToList();
+            htmlUris.ForEach(a => csUrisWithoutUnitTests.Add(a));
+
+            var stringRep = 0;
+
+            string regexPatternForInts = "[0-9]+\\.?[0-9]*";
+
+            foreach (var csFile in csUrisWithoutUnitTests)
+            {
+                var file = File.ReadAllText(csFile);
+
+                var yradRepetitions = CountRepetition(yardsToMeters, file, regexPatternForInts);
+                var inchRepetition = CountRepetition(inchToCentimeter, file, regexPatternForInts);
+                var mileRepetition = CountRepetition(mileToKilometer, file, regexPatternForInts);
+
+                if (yradRepetitions > VARIABLE_REPETITION_TRESHOLD)
+                {
+                    DuplicationEvidence.GiveEvidence($"Number {yardsToMeters} duplicated {yradRepetitions} times in file: {csFile}");
+                }
+                if (inchRepetition > VARIABLE_REPETITION_TRESHOLD)
+                {
+                    DuplicationEvidence.GiveEvidence($"Number {inchToCentimeter} duplicated {inchRepetition} times in file: {csFile}");
+                }
+                if (mileRepetition > VARIABLE_REPETITION_TRESHOLD)
+                {
+                    DuplicationEvidence.GiveEvidence($"Number {mileToKilometer} duplicated {mileRepetition} times in file: {csFile}");
+                }
+
+                stringRep += CountRepetition(stringCheck, file, stringCheck);
```

Comment:
I like this and I think the other checks should do it. Perhaps my proposed `CountRepeatedConstants` function should actually be like this:
```
private int CountRepeatedConstants(string valueToCheckAgainst)
{
    const string regexPatternForNumbers = "[0-9]+\\.?[0-9]*";
    return CountRepetition(valueToCheckAgainst, file, regexPatternForNumbers);
}
```
and then collect the sum and add the evidence at the end, like the string check does.